cpp/src/parquet/page_index.h

```diff
@@ -283,7 +280,8 @@ class PARQUET_EXPORT ColumnIndexBuilder {
   /// not write any data to the sink.
   ///
   /// \param[out] sink output stream to write the serialized message.
-  virtual void WriteTo(::arrow::io::OutputStream* sink) const = 0;
+  /// \param[in] encryptor encryptor to encrypt the serialized column index.
+  virtual void WriteTo(::arrow::io::OutputStream* sink, Encryptor* encryptor) const = 0;
```

Comment:
I'm curious why we don't make `encryptor` optional as in `PageIndexBuilder` below?

Reply:
It was suggested by @emkornfield in the review comments to avoid default parameter as much as possible. However, almost all encryptors/decryptors in the parquet-cpp are default to null. I think here I can make it default to null to be consistent.